analyzers/src/SonarAnalyzer.CSharp/Rules/LoopsAndLinq.cs

```diff
@@ -67,12 +67,17 @@ node switch
                 _ => null
             };
 
-        private static bool CanIfStatementBeMoved(IfStatementSyntax ifStatementSyntax) =>
-            ifStatementSyntax.Else == null
-            && ifStatementSyntax.Condition is InvocationExpressionSyntax invocationExpressionSyntax
-            && !invocationExpressionSyntax.DescendantNodes()
-                                          .OfType<ArgumentSyntax>()
-                                          .Any(argument => argument.RefOrOutKeyword.IsAnyKind(SyntaxKind.OutKeyword, SyntaxKind.RefKeyword));
+        private static bool CanIfStatementBeMoved(IfStatementSyntax ifStatementSyntax)
+        {
+            return ifStatementSyntax.Else == null && (ConditionIsPattern() || ConditionValidInvocation());
+
+            bool ConditionIsPattern() => IsPatternExpressionSyntaxWrapper.IsInstance(ifStatementSyntax.Condition);
+
+            bool ConditionValidInvocation() => ifStatementSyntax.Condition is InvocationExpressionSyntax invocationExpressionSyntax
+                                               && !invocationExpressionSyntax.DescendantNodes()
+                                               .OfType<ArgumentSyntax>()
```

Comment:
I am afraid that now the .OfType and .Any should be aligned with .DescendantNodes